src/Query.php

```diff
@@ -14,6 +14,58 @@ class Query
     protected $type;
     protected $wheres = [];
     protected $options = [];
+    protected $cacheQueryResults = false;
+    protected static $cachedRecords = [];
+
+    public static function eagerLoadAll(): Query
+    {
+        (new static)->cache()->get();
+
+        return new static;
```

Comment:
I've been using the `static::make()` method as a bit of a pattern. Not sure why, just because it was there I guess

Reply:
I've implemented this